### PR TITLE
Daico gareth

### PR DESCRIPTION
**Pull request by GreyEcologist**
issue I'm running into below:

Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: VM Exception while processing transaction: revert
    at Object.InvalidResponse (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/web3/lib/web3/errors.js:38:1)
    at /Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/web3/lib/web3/requestmanager.js:86:1
    at /Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/packages/truffle-migrate/index.js:225:1
    at /Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/packages/truffle-provider/wrapper.js:134:1
    at XMLHttpRequest.request.onreadystatechange (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/web3/lib/web3/httpprovider.js:128:1)
    at XMLHttpRequestEventTarget.dispatchEvent (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:64:1)
    at XMLHttpRequest._setReadyState (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:354:1)
    at XMLHttpRequest._onHttpResponseEnd (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:509:1)
    at IncomingMessage.<anonymous> (/Users/garethkmensah/DAOstack-Hackers-Kit/daico/node_modules/truffle/build/webpack:/~/xhr2/lib/xhr2.js:469:1)
    at IncomingMessage.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1094:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! daico@1.0.0 truffle-migrate: `truffle migrate`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the daico@1.0.0 truffle-migrate script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/garethkmensah/.npm/_logs/2018-11-12T19_17_52_379Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! daico@1.0.0 launch-local: `npm run load-arc && npm run webpack && npm run migrate-daostack && rm -rf build && npm run truffle-migrate && node dist/app.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the daico@1.0.0 launch-local script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

### CODE REVIEW

File: daico/contracts/ICOScheme.sol [hunk 1]
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.25;
+pragma solidity ^0.4.24;
```
Comment (orenyodfat) [inline daico/contracts/ICOScheme.sol]: why 24?

**Comment by orenyodfat** [comment]
@GreyEcologist what are the steps to reproduce ?
I see that in the migrations script you have no schemes..is that the intention ?

File: daico/contracts/ICOScheme.sol (hunk 1 above)
Comment (GreyEcologist) [reply daico/contracts/ICOScheme.sol]: I was running into an issue with truffle, so I tried different versions believing that might have been the difference. Basically `truffle migrate` kept throwing errors, so I tried downgrading the dependencies believing that might have been the cause to the errors. I'll re-add the original version as this doesn't seem to the cause of the issues. Currently still running into a "revert" issue every other time I run "truffle migrate" It usually goes away after I restart my machine. I've tried killing terminal and processes in the activity monitor but to no avail.

Comment (GreyEcologist) [reply daico/contracts/ICOScheme.sol]: The issue I'm currently working on, when I don't face the revert issue, is passing the avatar address from the deploy_dao.js. In the peepeth example the console.log called to get the avatar address is within a promise, wondering if this is why I'm not seeing the same address appear within the daico example.

File: daico/truffle.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ module.exports = {
           "https://kovan.infura.io/<YOUR-INFURA-KEY>"
         ),
       network_id: 42,
-      gasPrice: 1000000000 // 1 Gwei
+      gasPrice: 4712388 //1000000000 // 1 Gwei
```
Comment (ben-kaufman) [inline daico/truffle.js]: Please change the comment to match the new gas price.

File: daico/contracts/ICOScheme.sol (hunk 1 above)
Comment (ben-kaufman) [reply daico/contracts/ICOScheme.sol]: I'm not sure I quite understand what is the problem you are having with the Avatar. Can you please elaborate on that?

Comment (GreyEcologist) [reply daico/contracts/ICOScheme.sol]: So I need to capture the address of the avatar to pass it onto the index.js. However the console.log I have in the deploy.js don't return in the terminal nor in the google chrome developer console. I'm looking to capture the avatar address, but I'm not able to see it anywhere. I'm wondering if my console.log() calls are ever executed.

File: daico/migrations/2_deploy_dao.js [hunk 3]
```diff
@@ -40,23 +36,59 @@ module.exports = async function(deployer) {
...
-      foundersRep, // Founders initial reputation
+      foundersTokens,
+      foundersRep,
       0, // 0 because we don't use a UController
-      0, // no token cap
+      1000000,
       { gas: GAS_LIMIT }
     );
+
+    console.log("ABC 2");
```
[the remark this answers is not in the document]
Comment (GreyEcologist) [reply daico/migrations/2_deploy_dao.js]: Not done yet, so far I have not seen that console.log() being executed, wether the truffle migrate is successful or fail, this never gets executed, I'm wondering why.

**Pull request merged**